src/mcts/search.cc

```diff
@@ -2086,8 +2086,39 @@ void SearchWorker::FetchSingleNodeResult(NodeToProcess* node_to_process,
   }
   // For NN results, we need to populate policy as well as value.
   // First the value...
-  node_to_process->v = -computation.GetQVal(idx_in_computation);
-  node_to_process->d = computation.GetDVal(idx_in_computation);
+  auto v = -computation.GetQVal(idx_in_computation);
+  auto d = computation.GetDVal(idx_in_computation);
+  auto wdl_rescale_ratio = params_.GetWDLRescaleRatio();
+  auto wdl_rescale_diff = params_.GetWDLRescaleDiff();
+  if (wdl_rescale_ratio != 1.0f || wdl_rescale_diff != 0.0f) {
+    auto w = (1 + v - d) / 2;
+    auto l = (1 - v - d) / 2;
+    if (w > 0 && d > 0 && l > 0) {
+      auto a = FastLog(1 / l - 1);
+      auto b = FastLog(1 / w - 1);
+      auto s = 2 / (a + b);
+      auto mu = (a - b) / (a + b);
+      // Check whether root moves are from the set perspective.
+      bool root_stm = params_.GetPerspective() == "auto" ? true :
+            (params_.GetPerspective() == "white" ^
+             search_->played_history_.Last().IsBlackToMove());
+      auto sign = (root_stm ^ (node_to_process->depth & 1)) ? 1.0f : -1.0f;
+      auto s_new = s * std::sqrt(wdl_rescale_ratio);
+      auto mu_new = mu + sign * std::pow(s * 3.14159265, 2) / 3 *
+                     wdl_rescale_diff;
+      auto w_new = FastLogistic((-1.0f + mu_new) / s_new);
+      auto l_new = FastLogistic((-1.0f - mu_new) / s_new);
+      auto d_new = std::max(0.0f, 1.0f - w_new - l_new);
+      node_to_process->v = w_new - l_new;
+      node_to_process->d = d_new;
+    } else {
+      node_to_process->v = v;
+      node_to_process->d = d;
+    }
+  } else {
+    node_to_process->v = v;
+    node_to_process->d = d;
+  }
```

Comment:
I think that this new code should be a function, instead of an inlined block.

Reply:
I definitely agree! However, I am not yet worried about the optimization and code style, and only about the functionality. Also, I was wondering whether it is a good idea to do the conversion here, which is also executed on NN cache hits, instead of doing it in a way that the converted values are cached.

Reply:
I would like to see this conversion pushed "deeper" so that search is getting already adjusted `NNEval`s. This should also make merge with DAG easier.

Reply:
What do you consider "deeper"? Right now, it is as deep as it can get without touching the NN output itself; whenever a new node is created, it gets the (unaltered) WDL from the NN (either by an actual NN GPU eval or by a cache hit), and applies the conversion. Do you suggest to alter the NN WDL output, so we don't have to calculate the conversion multiple times on cache hits?

Reply:
Yes, I suggest to do the WDL adjustment before or on cache insertion so that we don't have to do that (multiple times) while fetching NN evals in search.

Reply:
I have thought about it a bit more, and I don't overly like storing it in the NN cache, as I can easily imagine restarting search with different parameters without clearing the cache. However, the conversion formula now really is lightweight, equivalent to calculating the PUCT score for maybe 5 nodes, but only performed once per addition of a new node. If there is any measureable CPU delay at all and (with DAG) the ratio of `(new nodes requesting NN evals) / (cache insertions)` is much bigger than 1, it is worth reconsidering, though that would likely mean touching multiple backends, and somehow transferring search parameters there.